src/app/events/scoap-squad/messageSetScoapRoles.ts

```diff
@@ -0,0 +1,31 @@
+import BotConversationMessageFlow from '../../service/scoap-squad/BotConversationMessageFlow';
+import { botConvoState } from '../../service/scoap-squad/ScoapDatabase';
+import { Message } from 'discord.js';
+import { scoapEmbedUpdate } from '../../service/scoap-squad/EditScoapDraft';
+
+export default async (message: Message): Promise<any> => {
+
+	const botConvo = botConvoState[message.author.id];
+	if (messageIsValid(message, botConvo)) {
+
+		if (botConvo.getEdit()) {
+			scoapEmbedUpdate(botConvo, message);
```

Comment:
my IDE is throwing that there is a missing `await`. Is this file suppose to run synchronously or async? Please also check lines 14,17

Reply:
ok, so upon testing, adding the awaits here causes some weird behavior. The new message notifications from the bot won't disappear anymore.

Reply:
Ok I think that weird behavior is happening because of the mismatch of missing await plus awaits. Basically EVERY async function should be called with an await. Why? Because the logic being processed for setting up roles and peopled required for scope squad should be complete in **order**. Unless the data retrieved from discord api or mongodb can be retrieved BEFORE asking the user for more information but I don't think that's happening here.

Let me know if this makes sense

Reply:
I think you're right :) Added awaits and seems to work good now.